### PR TITLE
improve the sorting rules of archives by using reverse year order

### PR DESCRIPTION
**Pull request by longjuan**
#### What type of PR is this?
/kind improvement
/area core

#### What this PR does / why we need it:
Display archives in reverse chronological order.
#### Which issue(s) this PR fixes:

Fixes https://github.com/halo-dev/halo/issues/3157

#### Special notes for your reviewer:
改进前：
image

改进后：
image

#### Does this PR introduce a user-facing change?

```release-note
归档以年份降序排序
```

### CODE REVIEW

**Review by ruibaby — COMMENTED** [review]
/lgtm

File: src/main/java/run/halo/app/theme/finders/impl/PostFinderImpl.java [hunk 1]
```diff
@@ -276,7 +276,7 @@ public Mono<ListResult<PostArchiveVo>> archives(Integer page, Integer size, Stri
                             .year(String.valueOf(key))
                             .months(monthArchives)
                             .build();
-                    }).toList();
+                    }).sorted(Comparator.comparing(PostArchiveVo::getYear).reversed()).toList();
```
Comment (guqing) [inline src/main/java/run/halo/app/theme/finders/impl/PostFinderImpl.java:279]: 可以把这个 sorted 令起一行

Comment (longjuan) [reply src/main/java/run/halo/app/theme/finders/impl/PostFinderImpl.java]: 已修改，https://github.com/longjuan/halo/commit/993522c613b8ff77563c73db4489aa72735fdaa0

**Review by ruibaby — APPROVED** [review]
/approve

**Review by JohnNiang — APPROVED** [review]
/lgtm

**Pull request merged**